overlord/snapstate/snapstate.go

```diff
@@ -618,24 +619,39 @@ func doUpdate(st *state.State, names []string, updates []*snap.Info, params func
 
 func applyAutoAliasesDelta(st *state.State, delta map[string][]string, op string, refreshAll bool, linkTs func(snapName string, ts *state.TaskSet)) (*state.TaskSet, error) {
 	applyTs := state.NewTaskSet()
+	kind := "refresh-aliases"
+	msg := i18n.G("Update aliases for snap %q")
+	if op == "prune" {
+		kind = "prune-auto-aliases"
+		msg = i18n.G("Prune automatic aliases for snap %q")
+	}
 	for snapName, aliases := range delta {
-		ts, err := resetAliases(st, snapName, aliases)
-		if err != nil {
+		if err := CheckChangeConflict(st, snapName, nil); err != nil {
 			if refreshAll {
 				// doing "refresh all", just skip this snap
 				logger.Noticef("cannot %s automatic aliases for snap %q: %v", op, snapName, err)
 				continue
 			}
 			return nil, err
 		}
+
+		snapsup := &SnapSetup{
+			SideInfo: &snap.SideInfo{RealName: snapName},
+		}
+		alias := st.NewTask(kind, fmt.Sprintf(msg, snapsup.Name()))
+		alias.Set("snap-setup", &snapsup)
+		if op == "prune" {
+			alias.Set("aliases", aliases)
+		}
+		ts := state.NewTaskSet(alias)
 		linkTs(snapName, ts)
 		applyTs.AddAll(ts)
 	}
 	return applyTs, nil
 }
 
-func autoAliasesUpdate(st *state.State, names []string, updates []*snap.Info) (new map[string][]string, mustRetire map[string][]string, transferTargets map[string]bool, err error) {
-	new, retired, err := AutoAliasesDelta(st, nil)
+func autoAliasesUpdate(st *state.State, names []string, updates []*snap.Info) (changed map[string][]string, mustPrune map[string][]string, transferTargets map[string]bool, err error) {
+	changed, dropped, err := autoAliasesDeltaV2(st, nil)
```

Comment:
Can we drop the V2 suffix in a follow up? I suppose after this PR those are all V2 already, right?